test-data/unit/check-modules.test

```diff
@@ -847,3 +849,17 @@ import x
 class Sub(x.Base):
     attr = x.Base.attr
 [out]
+
+[case testImportCycleStability3]
+import y
+[file x.py]
```

Comment:
I'd also suggest testing some additional things:

* Test every supported literal type.
* Test variables defined at module top level.
* Reveal the inferred type (to verify that it somehow isn't `Any`, for example).